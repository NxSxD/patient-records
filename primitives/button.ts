import styled from "styled-components";

export const Button = styled.button`
  height: 45px;
  padding: 0 1rem;
  font-size: 1.2rem;
  border: none;
  border-radius: 4px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  transition: background-color 250ms;
  &:hover {
    cursor: pointer;
    background-color: var(--primary-alt-color);
  }

  &:focus {
    outline: none;
  }
`;

export const ButtonAnchor = styled.a`
  height: 45px;
  padding: 0 1rem;
  font-size: 1.2rem;
  border: none;
  border-radius: 4px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  transition: background-color 250ms;
  &:hover {
    cursor: pointer;
    background-color: var(--primary-alt-color);
  }
`;

export const ButtonPrimary = styled(Button)`
  min-width: 120px;
  background-color: var(--primary-color);
  color: #fff;
`;

export const ButtonLink = styled(ButtonAnchor)`
  min-width: 120px;
  background-color: var(--primary-color);
  color: #fff;
`;
