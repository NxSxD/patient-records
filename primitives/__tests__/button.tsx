import React from "react";
import { render } from "@testing-library/react";
import { Button } from "../button";

describe("Button", () => {
  it("should render according to snapshot", () => {
    const { container } = render(<Button>Some Text</Button>);
    expect(container).toMatchSnapshot();
  });
});
