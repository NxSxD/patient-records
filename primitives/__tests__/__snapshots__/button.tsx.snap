// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render according to snapshot 1`] = `
.c0 {
  height: 45px;
  padding: 0 1rem;
  font-size: 1.2rem;
  border: none;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 250ms;
  transition: background-color 250ms;
}

.c0:hover {
  cursor: pointer;
  background-color: var(--primary-alt-color);
}

.c0:focus {
  outline: none;
}

<div>
  <button
    class="c0"
  >
    Some Text
  </button>
</div>
`;
